### PR TITLE
kubernetes: Add expiry-seconds flag to kubeconfig show and save commands.

### PR DESCRIPTION
**Pull request by andrewsomething**
Closes: https://github.com/digitalocean/doctl/issues/791

### CODE REVIEW

File: commands/kubernetes.go [hunk 1]
```diff
@@ -349,15 +360,22 @@ func kubernetesKubeconfig() *Command {
 
 	k8sCmdService := kubernetesCommandService()
 
-	CmdBuilder(cmd, k8sCmdService.RunKubernetesKubeconfigShow, "show <cluster-id|cluster-name>", "Show a Kubernetes cluster's kubeconfig YAML", `
+	cmdShowConfig := CmdBuilder(cmd, k8sCmdService.RunKubernetesKubeconfigShow, "show <cluster-id|cluster-name>", "Show a Kubernetes cluster's kubeconfig YAML", `
 This command prints out the raw YAML for the specified cluster's kubeconfig.	`, Writer, aliasOpt("p", "g"))
+	AddIntFlag(cmdShowConfig, doctl.ArgKubeConfigExpirySeconds, "", 0,
+		"The length of time the cluster credentials will be valid for in seconds. By default, the credentials expire after seven days.")
```
Comment (timoreimann) [inline commands/kubernetes.go:366]: The default is a little bit more involved for older cluster versions, but I think we can ignore that (or alternatively leave a pointer to [the relevant docs page](https://www.digitalocean.com/docs/kubernetes/how-to/connect-to-cluster/)).

**Review by timoreimann — COMMENTED** [review]
This is great, I know a few DOKS customers have been asking for supporting expiry on doctl. Thanks a ton Andrew.

File: commands/kubernetes.go [hunk 2]
```diff
@@ -128,11 +128,22 @@ type kubeconfigProvider struct {
...
+		kubeconfig, err = kube.GetKubeConfigWithExpiry(clusterID, int64(expirySeconds))
+		if err != nil {
+			return nil, err
+		}
+	default:
+		kubeconfig, err = kube.GetKubeConfig(clusterID)
+		if err != nil {
+			return nil, err
+		}
 	}
```
[the remark this answers is not in the document]
Comment (andrewsomething) [reply commands/kubernetes.go]: Good point!

**Review by timoreimann — APPROVED** [review]
LGTM. 👏

**Review by bentranter — APPROVED** [review]

**Pull request merged**